### ISSUE TITLE
Generated setup.py should use setuptools.setup instead of distutils.core.setup

### ISSUE THREAD

**Issue by seifertm** [sdispater/poetry#866]
- [x] I am on the [latest](https://github.com/sdispater/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/sdispater/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**: Gentoo Linux
- **Poetry version**: 0.12.11

The command `poetry build` generates a `setup.py` file based on the contents of `pyproject.toml`. The setup file imports `distutils.core.setup` as the setup function, but passes build arguments that are only available in `setuptools.setup`, such as `entry_points`. When the package is installed via pip, this is not an issue, because pip still uses `setuptools.setup`, regardless of what is imported (see the [distutils docs](https://docs.python.org/3/library/distutils.html)). However, other package managers may run into issues, because they use distutils. My distribution's package manager (Portage) generates warnings:
```
usr/lib64/python3.6/distutils/dist.py:261: UserWarning: Unknown distribution option: 'install_requires'
  warnings.warn(msg)
/usr/lib64/python3.6/distutils/dist.py:261: UserWarning: Unknown distribution option: 'entry_points'
  warnings.warn(msg)
/usr/lib64/python3.6/distutils/dist.py:261: UserWarning: Unknown distribution option: 'python_requires'
  warnings.warn(msg)
```
In this case, no scripts will be installed, because the `entry_points` argument is not applicable for distutils.

Since `setup.py` uses features that are explicitly part of setuptools, but not distutils, I suggest that `setuptools.setup` should be imported. Would you accept a PR for this?

**Comment by opoplawski** [comment]
This is becoming a problem in Fedora as well.  In particular we are seeing file based egg-info being generated rather than directories, which can cause issues with RPM.

**Comment by roniemartinez** [comment]
I just started testing `poetry` to see what's on it. Checked the generated `setup.py` and I didn't like it.

```python
from distutils.core import setup
```

Quoting from https://docs.python.org/3/library/distutils.html
> Most Python users will not want to use this module directly, but instead use the cross-version tools maintained by the Python Packaging Authority.

Sadly, I am one of these people. This issue should be a top priority.

**Comment by decathorpe** [comment]
For all fedora builds of python packages that were published with poetry, I have to patch the generated `setup.py` file to use setuptools instead of distutils, because otherwise the packages don't get built correctly. The default behaviour should totally be to use setup from setuptools in the template.

**Comment by pwwang** [comment]
Is there a workaround on this with `poetry` v0.12.16?

**Comment by eli-schwartz** [comment]
My solution was to create distro packages of "dephell" for the Arch Linux distribution and use https://dephell.readthedocs.io/cmd-deps-convert.html to convert pyproject.toml into a usable setup.py file. I just download the source tarball from GitHub instead of PyPI, and as a result I also get testsuite files.

**Comment by pwwang** [comment]
@eli-schwartz Thanks, this worked out!

 But I think the team still has to put this on higher priority.

**Comment by eli-schwartz** [comment]
Sure, happy to help. But you're correct that "don't use poetry, use something else" is not remotely optimal. I too am waiting for a better solution in poetry itself.

**Comment by techdragon** [comment]
I'm going to be including a fix for this in my work to fix #1350 since there is currently an outstanding PR ( https://github.com/pypa/setuptools/pull/1767 ) to bring `setuptools` in line with `wheel` and current documented practices regarding package license metadata, `pip` uses `setuptools` and I sincerely doubt there will be much traction trying to update `distutils`.  So using `setuptools` directly just make sense.
